lib/identity/hostdata/config_reader.rb

```diff
@@ -0,0 +1,106 @@
+module Identity
+  module Hostdata
+    class ConfigReader
+      attr_reader :app_root, :logger
+
+      def initialize(
+        app_root: Rails.root,
+        s3_client: nil,
+        logger: Logger.new(STDOUT)
+      )
```

Comment:
this is a hard rails dependency.... 😬 can we remove it? or nil it? maybe annotate we expect it to be an instance of `Pathname`?

```suggestion
      # @param [Pathname] app_root
      def initialize(
        app_root:,
        s3_client: nil,
        logger: Logger.new(STDOUT)
      )
```

Reply:
this is a good catch, I think it's fine to require passing it in explicitly from the caller